### PR TITLE
frdm-k28f: Initial support for the k28f MCU and frdm-k28f board

### PR DESCRIPTION
**Pull request by bergzand**
### Contribution description

This PR provides support for the k28f MCU and the frdm-k28f board.

I've intentionally configured the board to run the mcu at 120 MHz and not the full 150 MHz. For 150 MHz the MCU must run in the high speed mode which adds additional complexity when adding power modes. I suggest we add support for the high speed mode later after the PM support is merged.

### Testing procedure

A green murdock, and:

```
dist/tools/compile_and_test_for_board/compile_and_test_for_board.py frdm-k28f
```

Must work

### Issues/PRs references

None

### CODE REVIEW

**Comment by fjmolinas** [comment]
I don't have the board to test, do you mind posting testing results @bergzand?

File: boards/frdm-k28f/Makefile.features [hunk 1]
```diff
@@ -0,0 +1,13 @@
+CPU = kinetis
+CPU_MODEL = mk28fn2m0avmi15
+
+# Put defined MCU peripherals here (in alphabetical order)
+FEATURES_PROVIDED += arduino
```
Comment (aabadie) [inline boards/frdm-k28f/Makefile.features:5]: This line should be separated from the periph features. See how it's done for other boards providing the arduino feature.

File: boards/frdm-k28f/include/adc_params.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ */
+
+/**
+ * @ingroup   boards_frdm-k28f
+ * @{
+ *
+ * @file
+ * @brief     Board specific configuration of direct mapped ADC
+ *
+ * @author    Joakim Nohlgård <joakim.nohlgard@eistec.se>
```
Comment (aabadie) [inline boards/frdm-k28f/include/adc_params.h:18]: Add yourself to the list of authors :)

**Comment by fjmolinas** [comment]
ping @bergzand!

**Comment by bergzand** [comment]
Results so far:
```
Failures during test:
- [tests/driver_at86rf2xx_aes](tests/driver_at86rf2xx_aes/test.failed)
- [tests/emcute](tests/emcute/test.failed)
- [tests/gnrc_dhcpv6_client](tests/gnrc_dhcpv6_client/test.failed)
- [tests/gnrc_dhcpv6_client_6lbr](tests/gnrc_dhcpv6_client_6lbr/test.failed)
- [tests/gnrc_ipv6_ext](tests/gnrc_ipv6_ext/test.failed)
- [tests/gnrc_ipv6_ext_frag](tests/gnrc_ipv6_ext_frag/test.failed)
- [tests/gnrc_ipv6_ext_opt](tests/gnrc_ipv6_ext_opt/test.failed)
- [tests/gnrc_ipv6_nib_dns](tests/gnrc_ipv6_nib_dns/test.failed)
- [tests/gnrc_rpl_srh](tests/gnrc_rpl_srh/test.failed)
- [tests/gnrc_sock_dns](tests/gnrc_sock_dns/test.failed)
- [tests/gnrc_tcp](tests/gnrc_tcp/test.failed)
- [tests/malloc](tests/malloc/test.failed)
- [tests/periph_flashpage](tests/periph_flashpage/test.failed)
- [tests/periph_timer](tests/periph_timer/test.failed)
- [tests/pkg_libfixmath_unittests](tests/pkg_libfixmath_unittests/test.failed)
- [tests/pkg_littlefs](tests/pkg_littlefs/test.failed)
- [tests/pkg_littlefs2](tests/pkg_littlefs2/test.failed)
- [tests/pkg_semtech-loramac](tests/pkg_semtech-loramac/test.failed)
- [tests/pkg_spiffs](tests/pkg_spiffs/test.failed)
- [tests/unittests](tests/unittests/test.failed)
```

**Comment by aabadie** [comment]
The following tests seems problematic:

```
- [tests/malloc](tests/malloc/test.failed)
- [tests/periph_flashpage](tests/periph_flashpage/test.failed)
- [tests/periph_timer](tests/periph_timer/test.failed)
- [tests/unittests](tests/unittests/test.failed)
```

Can you paste the output ? Or do you have an idea what could be the cause ?

**Comment by fjmolinas** [comment]
> - [tests/periph_flashpage](tests/periph_flashpage/test.failed)

Is the bootloader image alined?

**Comment by bergzand** [comment]
> - [tests/periph_timer](tests/periph_timer/test.failed)

The third timer config is broken.

**Comment by bergzand** [comment]
The filesystem failures are related to a misconfigured MTD device I think. The filesystem example is also broken

File: cpu/kinetis/Makefile.features [hunk 3]
```diff
@@ -1,11 +1,16 @@
 FEATURES_PROVIDED += periph_cpuid
 FEATURES_PROVIDED += periph_pm
 
-# TRNG driver is not implemented for mkw41z models
+# TRNG driver is not implemented for mkw41z and the mk28f models
 _KINETIS_CPU_MODELS_WITHOUT_HWRNG += mkw41z256vht4 mkw41z512vht4
+_KINETIS_CPU_MODELS_WITHOUT_HWRNG += kk28fn2m0cau15
```
Comment (benpicco) [inline cpu/kinetis/Makefile.features:6]: Would ` kk28fn2%` also work here or are there some parts with HWRNG?

File: cpu/kinetis/periph/timer.c [hunk 4]
```diff
@@ -32,7 +32,7 @@
 #include "periph_conf.h"
 #include "periph/timer.h"
 
-#ifdef PIT_LTMR64H_LTH_MASK
+#if defined(PIT_LTMR64H_LTH_MASK) && !defined(KINETIS_PIT_COMBINED_IRQ)
```
Comment (benpicco) [inline cpu/kinetis/periph/timer.c:35]: Is the comment below still valid?

**Review by benpicco — COMMENTED** [review]
looks good, but what's up with malloc?
(The links to the test results don't work anymore)

File: boards/frdm-k28f/mtd.c [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    .wait_chip_erase = 40LU * US_PER_SEC,
+    .wait_32k_erase = 100LU *US_PER_MS,
+    .wait_sector_erase = 150LU * US_PER_MS,
+    .wait_4k_erase = 50LU * US_PER_MS,
+    .wait_chip_wake_up = 30LU * US_PER_MS,
+    .clk = FRDM_K28F_NOR_SPI_CLK,
+    .flag = FRDM_K28F_NOR_FLAGS,
+    .spi = FRDM_K28F_NOR_SPI_DEV,
+    .mode = FRDM_K28F_NOR_SPI_MODE,
+    .cs = FRDM_K28F_NOR_SPI_CS,
```
Comment (benpicco) [inline boards/frdm-k28f/mtd.c:41]:
```suggestion
    .cs = FRDM_K28F_NOR_SPI_CS,
    .wp = GPIO_UNDEF,
    .hold = GPIO_UNDEF,
```

**Comment by aabadie** [comment]
Needs a rebase

**Pull request closed without merging**